staging/src/k8s.io/apiserver/pkg/audit/context_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"testing"
+)
+
+func TestWithAuditAnnotations(t *testing.T) {
+	var annotations []annotation
+	expContext := context.WithValue(context.Background(), auditAnnotationsKey, &annotations)
+
+	// Add key auditAnnotations to context
+	gotContext := WithAuditAnnotations(context.Background())
+	assert.Equal(t, expContext, gotContext)
+
+	// Key auditAnnotations is already in context
+	gotContext = WithAuditAnnotations(gotContext)
+	assert.Equal(t, expContext, gotContext)
+}
+
+func TestAddAuditAnnotation(t *testing.T) {
+	testCases := []struct {
+		desc              string
+		expCtxAnnotations []annotation
+		getFunc           func(t *testing.T) []annotation
+	}{
+		{
+			desc:              "auditEvent in context",
+			expCtxAnnotations: ([]annotation)(nil),
+			getFunc: func(t *testing.T) []annotation {
+				ev := &audit.Event{}
+				// add auditEvent to context,
+				ctx := request.WithAuditEvent(context.Background(), ev)
+
+				// record annotations to auditEvent not context
+				AddAuditAnnotation(ctx, "foo", "bar")
```

Comment:
you should also check the event to see that the expected action was in fact performed

Reply:
Thanks for your review.
Done.

Reply:
@stlaz Can you review again?